src/site/content/en/blog/building-a-breadcrumbs-component/index.md

```diff
@@ -0,0 +1,498 @@
+Each `.crumb` also establishes a horizontal vertically aligned layout with some
+gap, but specially targets its link children and specifies the style
+`white-space: nowrap`. This is crucial for multi-word breadcrumbs as we dont
+want them to go multi-line. Later in this post we'll add styles to handle the
+horizontal overflow this `white-space` property caused. 
+
+```css
+.crumb {
+  display: inline-flex;
+  align-items: center;
+  gap: calc(var(--nav-gap) / 4);
+
+  & > a {
+    white-space: nowrap;
+
+    &[aria-current="page"] {
+      font-weight: bold;
+    }
+  }
+}
+```
+
+`aria-current="page"` is added to help the current page link stand out from the
+rest. Not only will screen reader users have a clear indicator that the link is
+for the current page, we've visually styled the element to help sighted users
+get a similar user experience.
+
+The `.crumbicon` component uses grid to stack an SVG icon with a "nearly
+invisible" `<select>` element.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/jx2uAiFT8NEHD5AbuSp3.png",
+alt="grid devtools shown overlaying a button where the row and column are both
+named stack", width="218", height="180" %}
+
+```css/8
+.crumbicon {
+  --crumbicon-size: 3ch;
+
+  display: grid;
+  grid: [stack] var(--crumbicon-size) / [stack] var(--crumbicon-size);
+  place-items: center;
+
+  & > * {
+    grid-area: stack;
+  }
+}
+```
+
+The `<select>` element is last in the DOM, so it's on top of the stack, and interactive. Add a style of `opacity: .01` so the element is still usable, and the result is a select box that fits perfectly to the shape of the icon. This is a nice way to customize the look of a select element while maintaining the built-in functionality.
+
+```css
+.disguised-select {
+  inline-size: 100%;
+  block-size: 100%;
+  opacity: .01;
+  font-size: 16px; /* fix iOS zoom */
+}
+```
+
+{% Aside %}
+Try using this icon component from mobile!
+{% endAside %}
+
+{% Video
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/n3MsbELMwsvYn355hXdi.mp4",
+  class="w-screenshot",
+  autoplay="true",
+  loop="true",
+  muted="true"
+%}
+
+#### Overflow
+
+Breadcrumbs should be able to represent a very long trail. I'm a fan of allowing
+things to go offscreen horizontally, when appropriate, and I felt this
+breadcrumbs component qualified well. 
+
+```css
+.breadcrumbs {
+  overflow-x: auto;
+  overscroll-behavior-x: contain;
+  scroll-snap-type: x proximity;
+  scroll-padding-inline: calc(var(--nav-gap) / 2);
+
+  & > .crumb:last-of-type {
+    scroll-snap-align: end;
+  }
+
+  @supports (-webkit-hyphens:none) { & {
+    scroll-snap-type: none;
+  }}
+}
+```
+
+The overflow styles set up the following UX:
+
+1. Horizontal scroll with overscroll containment.
+1. Horizontal scroll padding.
+1. One snap point on the last crumb. This means that on page load the first
+   crumb loads snapped and in view.
+1. Removes the snap point from Safari, which struggles with the horizontal
+   scrolling and snap effect combinations.
```

Comment:
```suggestion
- Horizontal scroll with overscroll containment.
- Horizontal scroll padding.
- One snap point on the last crumb. This means that on page load the first
   crumb loads snapped and in view.
- Removes the snap point from Safari, which struggles with the horizontal
   scrolling and snap effect combinations.
```

Reply:
@rachelandrew Reserve numbered lists for sequences of steps. Here's the styleguide reference: https://developers.google.com/style/lists?hl=en#sub-steps-in-a-numbered-procedure